### PR TITLE
Update AprilTag with upstream

### PR DESCRIPTION
**Pull request by s-trinh**
Manually sync AprilTag with upstream repo @ [ee28e35](https://github.com/AprilRobotics/apriltag/commit/ee28e35911ca0ccdc552c22ee13d9a72c71d5dd4)

---

The second commit has big changes (see https://github.com/AprilRobotics/apriltag/commit/a29f685ff40564e951f3ea276e518e7d541cdd05). Hopefully it will not lead to some regression.

### CODE REVIEW

File: 3rdparty/apriltag/apriltag_quad_thresh.cpp [hunk 1]
```diff
@@ -954,7 +957,7 @@ int fit_quad(
 
         area += sqrt(p*(p-length[0])*(p-length[1])*(p-length[2]));
 
-        if (area < tag_width*tag_width) {
+        if (area < 0.95*tag_width*tag_width) {
```
Comment (s-trinh) [inline 3rdparty/apriltag/apriltag_quad_thresh.cpp:960]: See: https://github.com/AprilRobotics/apriltag/commit/e0ba863c65589e551e9d8647874a83f39978b473

File: 3rdparty/apriltag/apriltag.cpp [hunk 2]
```diff
@@ -654,8 +654,15 @@ float quad_decode(apriltag_detector_t* td, apriltag_family_t *family, image_u8_t
...
-    graymodel_solve(&blackmodel);
+    if (family->width_at_border > 1) {
+        graymodel_solve(&whitemodel);
+        graymodel_solve(&blackmodel);
+    } else {
+        graymodel_solve(&whitemodel);
+        blackmodel.C[0] = 0;
+        blackmodel.C[1] = 0;
+        blackmodel.C[2] = blackmodel.B[2]/4;
+    }
```
Comment (s-trinh) [inline 3rdparty/apriltag/apriltag.cpp:665]: See: https://github.com/AprilRobotics/apriltag/commit/43dbab7950939629b1793319d08e22a0266807d4

File: 3rdparty/apriltag/apriltag.cpp [hunk 3]
```diff
@@ -1455,7 +1465,7 @@ image_u8_t *apriltag_to_image(apriltag_family_t *fam, int idx)
     image_u8_t *im = image_u8_create(fam->total_width, fam->total_width);
 
     int white_border_width = fam->width_at_border + (fam->reversed_border ? 0 : 2);
-    int white_border_start = (fam->total_width - fam->width_at_border)/2;
+    int white_border_start = (fam->total_width - white_border_width)/2;
```
Comment (s-trinh) [inline 3rdparty/apriltag/apriltag.cpp:1468]: See: https://github.com/AprilRobotics/apriltag/commit/1298c78afe78f603940a6ffc4d72b8ddd992d94a

File: 3rdparty/apriltag/apriltag_pose.cpp [hunk 4]
```diff
@@ -101,6 +101,12 @@ double orthogonal_iteration(matd_t** v, matd_t** p, matd_t** t, matd_t** R, int
         matd_destroy(M3);
         matd_destroy(*R);
         *R = matd_op("M*M'", M3_svd.U, M3_svd.V);
+        double R_det = matd_det(*R);
+        if (R_det < 0) {
+            matd_put(*R, 0, 2, - matd_get(*R, 0, 2));
+            matd_put(*R, 1, 2, - matd_get(*R, 1, 2));
+            matd_put(*R, 2, 2, - matd_get(*R, 2, 2));
+        }
```
Comment (s-trinh) [inline 3rdparty/apriltag/apriltag_pose.cpp:109]: See: https://github.com/AprilRobotics/apriltag/commit/f409268c918ff0b05aa495628160c133baeedfb7

File: 3rdparty/apriltag/apriltag_quad_thresh.cpp [hunk 5]
```diff
@@ -380,8 +380,11 @@ int quad_segment_maxima(apriltag_detector_t *td, zarray_t *cluster, struct line_
     free(errs);
 
     // if we didn't get at least 4 maxima, we can't fit a quad.
-    if (nmaxima < 4)
+    if (nmaxima < 4){
+        free(maxima);
+        free(maxima_errs);
```
Comment (s-trinh) [inline 3rdparty/apriltag/apriltag_quad_thresh.cpp:385]: See: https://github.com/AprilRobotics/apriltag/commit/5ed87e741c9c61167a056676f9c69c544516410e

File: 3rdparty/apriltag/common/matd.cpp [hunk 6]
```diff
@@ -1110,8 +1110,10 @@ static matd_svd_t matd_svd_tall(matd_t *A, int flags)
             double mag = sqrt(mag2);
 
             // this case can occur when the vectors are already perpendicular
-            if (mag == 0)
+            if (mag == 0) {
+                free(v);
```
Comment (s-trinh) [inline 3rdparty/apriltag/common/matd.cpp:1114]: See: https://github.com/AprilRobotics/apriltag/commit/d0252946999fbcaf79504073acacdd079613fb26

File: 3rdparty/apriltag/common/matd.cpp [hunk 7]
```diff
@@ -1144,10 +1146,10 @@ static matd_svd_t matd_svd_tall(matd_t *A, int flags)
...
     }
 
-    // maxiters used to be smaller to prevent us from looping forever,
-    // but this doesn't seem to happen any more with our more stable
-    // svd22 implementation.
-    int maxiters = 1UL << 30;
+    // empirically, we find a roughly linear worst-case number of iterations
+    // as a function of rows*cols. maxiters ~= 1.5*nrows*ncols
+    // we're a bit conservative below.
+    int maxiters = 200 + 2 * A->nrows * A->ncols;
```
Comment (s-trinh) [inline 3rdparty/apriltag/common/matd.cpp:1152]: See: https://github.com/AprilRobotics/apriltag/commit/ae6f7c47d9476329e26631571ecb57c13745f764

**Comment by fspindle** [comment]
@s-trinh Thanks a lot.

This PR should fix #986

**Pull request merged**